WebRoot/js/zimbraMail/calendar/view/ZmCalendarPrefsPage.js

```diff
@@ -864,60 +792,61 @@ ZmCustomWorkHoursDlg.prototype.initialize = function(workHours) {
         inputTime = new Date();
         inputTime.setHours(workHours[i].startTime/100, workHours[i].startTime%100, 0);
         startTimeSelect.set(inputTime);
-        startTimeSelect.setEnabled(workHours[i].isWorkingDay);
-        this._startTimeSelect.push(startTimeSelect);
-
+        startTimeSelect.setEnabled(isCustom ? workHours[i].isWorkingDay : false);
+        this._startTimeSelectCustom.push(startTimeSelect);
 
         inputTime = new Date();
         inputTime.setHours(workHours[i].endTime/100, workHours[i].endTime%100, 0);
         el = document.getElementById(this._htmlElId + "_CAL_WORKING_END_TIME_"+i);
         endTimeSelect = new DwtTimeInput(this, DwtTimeInput.END, el);
         endTimeSelect.set(inputTime);
-        endTimeSelect.setEnabled(workHours[i].isWorkingDay);
-        this._endTimeSelect.push(endTimeSelect);
+        endTimeSelect.setEnabled(isCustom ? workHours[i].isWorkingDay : false);
+        this._endTimeSelectCustom.push(endTimeSelect);
 
+        label = new DwtLabel({parent:this, parentElement:(this._htmlElId + "_CAL_WORKING_DAY_LABEL_" + i)});
+        label.setText(AjxDateUtil.WEEKDAY_MEDIUM[i]);
+        label.setEnabled(isCustom ? workHours[i].isWorkingDay : false);
+        this._workDayCustom.push(label);
 
-        checkbox = new DwtCheckbox({parent:this, parentElement:(this._htmlElId + "_CAL_WORKING_DAY_" + i)});
-        checkbox.setText(AjxDateUtil.WEEKDAY_MEDIUM[i]);
-	    checkbox.setSelected(workHours[i].isWorkingDay);
-        checkbox.addSelectionListener(new AjxListener(this, this._setTimeInputEnabled, [i, checkbox]));
-        this._workDaysCheckBox.push(checkbox);
+        Dwt.condClass(Dwt.byId(this._htmlElId + "_CAL_SEP_LABEL_" + i), isCustom && workHours[i].isWorkingDay, DwtControl.NORMAL, DwtControl.DISABLED); //toggle state of `to` label
     }
 };
 
-ZmCustomWorkHoursDlg.prototype.reset =
+ZmWorkHours.prototype.customWorkHoursReset =
 function() {
     var i,
-    inputTime;
+    inputTime,
+    isCustom;
 
+    isCustom = this._radioCustom.isSelected(); //if "custom" radio-button is selected
```

Comment:
Yes, we can but any specific reason for that? Because I see in this method practise of declaring variable first before assignment. And it's just in-line with that.  https://github.com/Zimbra/zm-web-client/blob/775abaa15435f2af4fed69f03441b88bc70ef346/WebRoot/js/zimbraMail/calendar/view/ZmCalendarPrefsPage.js#L817

Let me know.